crates/primitives/src/transaction/access_list.rs

```diff
@@ -48,6 +58,14 @@ impl AccessList {
             })
             .collect()
     }
+
+    /// Calculates a heuristic for the in-memory size of the [AccessList].
+    #[inline]
+    pub fn size(&self) -> usize {
+        // take into account capacity
+        self.0.iter().map(AccessListItem::size).sum::<usize>() +
+            (self.0.capacity() - self.0.len()) * mem::size_of::<AccessListItem>()
```

Comment:
Why do we subtract len from capacity?

Reply:
This is so we can account for the entire `Vec` in the heuristic, for example if the vec was initialized using `with_capacity`, it will still allocate, but those bytes won't be counted by the call to `iter`. I don't think we do this much, especially since we do lots of shrinking now (from the last OOM), but it seems more correct

Reply:
hmm, i might be wrong about this, we might not want to subtract len from capacity here - at least an example from `DeepSizeOf` does it differently https://docs.rs/deepsize/latest/deepsize/trait.DeepSizeOf.html#tymethod.deep_size_of_children

Reply:
changed so we no longer do this